salt/_states/metalk8s_checks.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+"""
+State module for handling MetalK8s checks.
+"""
+
+from salt.exceptions import CheckError
+
+__virtualname__ = "metalk8s_checks"
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def module_run(name, **kwargs):
+    """Wrap `module.run` state, pretty print the error and fail gracefully if
+    the module raises a `CheckError` exception.
```

Comment:
Minor: Mention also that this state should really do NOT changes anything on the host it's pure "checks"

Reply:
Done, I've also put a little explanation why we're using a state to do this.